sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/InternalRow.scala

```diff
@@ -27,6 +27,63 @@ import org.apache.spark.sql.catalyst.expressions.GenericRow
 abstract class InternalRow extends Row {
   // A default implementation to change the return type
   override def copy(): InternalRow = {this}
+
+  // A default version (slow), used for tests
```

Comment:
Is this really only used for tests?  Won't it be used when we use an internal row as a HashKey?